compiler/rustc_monomorphize/src/errors.rs

```diff
@@ -0,0 +1,74 @@
+use std::path::PathBuf;
+
+use rustc_errors::ErrorGuaranteed;
+use rustc_macros::SessionDiagnostic;
+use rustc_session::SessionDiagnostic;
+use rustc_span::Span;
+
+#[derive(SessionDiagnostic)]
+// FIXME: This should be #[fatal...], but that doesn't exist, yet...
```

Comment:
I think `error()` is still correct here -- choosing how to emit the diagnostic (i.e. as an error or as a fatal) is up to the user actually calling `emit_*` on this.

Reply:
I don't completely understand how diagnostics are handled, but errors and fatals have different type signatures. When errors use ErrorGuaranteed, fatals use !(never). I think that's why existing functions have a separate version for fatals.

My PR(#100694) adds #[fatal(..)], which should be helpful here.

Reply:
This pull request will need to wait for #100694 to land so that it can use `#[fatal(..)]`.